src/Icicle/Avalanche/Prim/Eval.hs

```diff
@@ -169,5 +201,35 @@ evalPrim p vs
    | otherwise
    = List.drop 1 (xs <> [x])
 
+  uz (fs, ss) (VPair f s)
+   = return (f     : fs, s     : ss)
+  uz (fs, ss) VUnit -- unitialised array
+   = return (VUnit : fs, VUnit : ss)
+  uz _ _
+   = primError
+
+  us (bs, ls, rs) (VLeft l)
+   = return ( VBool  False  : bs
+            , l             : ls
+            , VUnit         : rs )
```

Comment:
should probably use defaultOfType instead of VUnit. I don't think we have partial evaluation on Avalanche yet, but if we did, VUnit would change the type